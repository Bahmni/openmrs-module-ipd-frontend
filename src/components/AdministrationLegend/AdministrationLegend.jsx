import React from "react";
import AdministeredIcon from "../../icons/administered.svg";

import AdministeredLateIcon from "../../icons/administered-late.svg";
import LateIcon from "../../icons/late.svg";
import NotAdministeredIcon from "../../icons/not-administered.svg";
import PendingIcon from "../../icons/pending.svg";
import StoppedIcon from "../../icons/stopped.svg";
import "./AdministrationLegend.scss";
import { FormattedMessage } from "react-intl";

export default function AdministrationLegend() {
  return (
    <div className={"drug-chart-legend"}>
      <div>
        <PendingIcon />
        <FormattedMessage id="PENDING" defaultMessage={"Pending"} />
      </div>
      <div>
        <LateIcon />
        <FormattedMessage id="LATE" defaultMessage={"Late"} />
      </div>
      <div>
        <AdministeredIcon />
        <FormattedMessage id="ADMINISTERED" defaultMessage={"Administered"} />
      </div>
      <div>
        <AdministeredLateIcon />
        <FormattedMessage
          id="ADMINISTERED_LATE"
          defaultMessage={"Administered Late"}
        />
      </div>
      <div>
        <NotAdministeredIcon />
        <FormattedMessage
          id="NOT_ADMINISTERED"
          defaultMessage={"Not Administered"}
        />
      </div>
      <div>
        <StoppedIcon />
        <FormattedMessage id="STOPPED" defaultMessage={"Stopped"} />
      </div>
    </div>
  );
}
