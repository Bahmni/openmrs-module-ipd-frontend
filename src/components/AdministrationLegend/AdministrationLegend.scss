.drug-chart-legend {
  display: flex;
  align-items: center;
  gap: 15px;
  padding: 10px;
  font-size: 12px;
  flex-wrap: wrap;
  > * {
    display: flex;
    align-items: center;
    gap: 5px;
  }
}
