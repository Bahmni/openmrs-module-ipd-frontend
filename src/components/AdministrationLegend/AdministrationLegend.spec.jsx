import { render } from "@testing-library/react";
import React from "react";
import AdministrationLegend from "./AdministrationLegend";

describe("AdministrationLegend", () => {
  it("should render without crashing", () => {
    const { container } = render(<AdministrationLegend />);
    expect(container).toMatchSnapshot();
  });
  it("should render legends", () => {
    const { getByText } = render(<AdministrationLegend />);
    expect(getByText("Pending")).toBeTruthy();
    expect(getByText("Late")).toBeTruthy();
    expect(getByText("Administered")).toBeTruthy();
    expect(getByText("Administered Late")).toBeTruthy();
    expect(getByText("Not Administered")).toBeTruthy();
  });
});
