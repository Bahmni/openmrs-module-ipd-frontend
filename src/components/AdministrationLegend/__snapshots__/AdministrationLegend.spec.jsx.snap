// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdministrationLegend should render without crashing 1`] = `
<div>
  <div
    class="drug-chart-legend"
  >
    <div>
      <test-file-stub />
      Pending
    </div>
    <div>
      <test-file-stub />
      Late
    </div>
    <div>
      <test-file-stub />
      Administered
    </div>
    <div>
      <test-file-stub />
      Administered Late
    </div>
    <div>
      <test-file-stub />
      Not Administered
    </div>
    <div>
      <test-file-stub />
      Stopped
    </div>
  </div>
</div>
`;
